server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -86,7 +86,11 @@ public static XContentParser createParser(
         BytesReference bytes,
         XContentType xContentType
     ) throws IOException {
-        return createParser(XContentParserConfiguration.EMPTY.withRegistry(registry).withDeprecationHandler(deprecation), bytes);
+        return createParser(
+            XContentParserConfiguration.EMPTY.withRegistry(registry).withDeprecationHandler(deprecation),
+            bytes,
+            xContentType
```

Comment:
arg good catch!

Reply:
given that you fixed content type auto guessing here, I wonder if the switch from registry to configuration is still needed. I must admit I am not familiar with the configuration though so I may totally be missing something...

Reply:
I guess it's not technically "needed" but it moves us away from a deprecated API and saves some (though probably irrelevant) objects by not having to set up the config on every call.